pkg/task/serialize.go

```diff
@@ -51,6 +58,42 @@ func SerializeHttpPayload(ctx echo.Context) (*types.TaskPayload, error) {
 		// Remaining payload is treated as kwargs if not empty
 		taskPayload.Kwargs = payload
 	}
+}
 
-	return taskPayload, nil
+func parseRequestArgs(queryParams url.Values, taskPayload *types.TaskPayload) {
+	if taskPayload.Kwargs == nil {
+		taskPayload.Kwargs = make(map[string]interface{})
+	}
+
+	for key, values := range queryParams {
+		if len(values) == 1 {
+			taskPayload.Kwargs[key] = convertToNumericIfPossible(values[0])
+		} else {
+			allFloat := true
+			floats := make([]float64, len(values))
+
+			for i, value := range values {
+				convertedValue := convertToNumericIfPossible(value)
+				switch v := convertedValue.(type) {
+				case float64:
+					floats[i] = v
+				default:
+					allFloat = false
+					continue
+				}
+			}
+			if allFloat {
+				taskPayload.Kwargs[key] = floats
+			} else {
+				taskPayload.Kwargs[key] = values
+			}
+		}
+	}
+}
+
+func convertToNumericIfPossible(s string) interface{} {
```

Comment:
This is probably a personal preference and a nitpick, but I'm not a fan of the "IfPossible" part lol. What about naming this to something like `tryParseNumeric`?

Reply:
yeah that is fair lol. I will change it to `tryParseNumeric`